tls-client/main.cpp

```diff
@@ -210,29 +207,39 @@ class HelloHTTPS {
         ret = _tcpsocket->connect(_domain, _port);
         if (ret != NSAPI_ERROR_OK) {
             mbedtls_printf("Failed to connect\r\n");
-            onError(_tcpsocket, -1);
+            printf("MBED: Socket Error: %d\r\n", ret);
+            _tcpsocket->close();
```

Comment:
Hence instead I called it at each exit point. This applies to all similar comments.